ofrak_components/ofrak_components/zstd.py

```diff
@@ -0,0 +1,96 @@
+import subprocess
+import tempfile
+from dataclasses import dataclass
+
+from ofrak import Packer, Unpacker, Resource
+from ofrak.component.packer import PackerError
+from ofrak.component.unpacker import UnpackerError
+from ofrak.core import (
+    GenericBinary,
+    format_called_process_error,
+    MagicMimeIdentifier,
+    MagicDescriptionIdentifier,
+)
+from ofrak.model.component_model import CC, ComponentConfig
+from ofrak_type.range import Range
+
+
+class ZstdData(GenericBinary):
+    """
+    A zstd binary blob.
+    """
+
+    async def get_child(self) -> GenericBinary:
+        return await self.resource.get_only_child_as_view(GenericBinary)
+
+
+@dataclass
+class ZstdPackerConfig(ComponentConfig):
+    compression_level: int
```

Comment:
Out of scope for this PR, but I believe that packers and unpackers should not use configs. Instead, they should use attributes of the resource when possible. For example the flash components use resource attributes instead of configs.

https://github.com/redballoonsecurity/ofrak/blob/4c9723e23f9b22a7f745d469354174a4f3be0f3e/ofrak_components/ofrak_components/flash.py#L77-L90

https://github.com/redballoonsecurity/ofrak/blob/4c9723e23f9b22a7f745d469354174a4f3be0f3e/ofrak_components/ofrak_components/flash.py#L313-L314